src/DataManagement/Goals/GoalManager.cs

```diff
@@ -19,59 +22,116 @@ public GoalManager(IGoalList goalList, int userId = 0)
 
         public bool Add(IGoal goal)
         {
-            var id = GoalDbUpdater.Add(goal, UserId);
+            var id = GoalDatabase.Create(goal, UserId);
             //Check if id is correct, return false if something is wrong
-
             goal.Id = id;
-            goal.UserId = UserId;
             GoalList.Add(goal);
-            return false;
+            return true;
+        }
+
+        public bool AddRange(IGoalList goalList)
+        {
+            if (!GoalDatabase.CreateList(goalList, UserId))
+            {
+                return false;
+            }
+
+            GoalList.AddRange(goalList);
+            return true;
+        }
+
+        public bool Edit(IGoal oldGoal, IGoal updatedGoal)
+        {
+            return Edit(oldGoal.Id, updatedGoal);
         }
 
-        public bool Edit(IGoal oldGoal, IGoal newGoal)
+        public bool Edit(int id, IGoal updatedGoal)
         {
-            if (!GoalDbUpdater.Edit(oldGoal, newGoal))
+            if (!GoalDatabase.Update(id, UserId, updatedGoal))
             {
                 return false;
             }
-            var localGoal = GoalList.FirstOrDefault(x => x.Id == oldGoal.Id);
+            var localGoal = GoalList.FirstOrDefault(x => x.Id == id);
             if (localGoal is null)
             {
-                ExceptionHandler.Log("Edited goal id: " + oldGoal.Id + " in database but couldn't find it locally");
+                ExceptionHandler.Log("Edited goal id: " + id + " in database but couldn't find it locally");
```

Comment:
Just a thought: this code is checked with if and will not raise exception of any sort, so it is kinda strange to use ExceptionHandler when there is no really no exception (it is controlled with control flow).

Reply:
Good point actually. What I wanted to do was to know if something went wrong at this place in code, and the exception handler seemed like the way to do it? Maybe I should actually throw an exception here if something goes wrong, and surround it with a try catch block so we can actually print stack trace and all that?